Assets/Scripts/UI/UIAnimationController.cs

```diff
@@ -0,0 +1,64 @@
+using UnityEngine;
+using UnityEngine.EventSystems;
+
+public class UIAnimationController : MonoBehaviour, IPointerEnterHandler, IPointerExitHandler
+{
+    private Animator animator;
+    private static UIAnimationController currentlyHoveredButton;
+
+    private void Awake()
+    {
+        animator = GetComponent<Animator>();
```

Comment:
should add a requirescomponent to the class at the top if it requires this or check for not null at the very least.